File: legacy/sdk/python/axion/resources/memory.py
"""Memory resource."""
from __future__ import annotations

from typing import TYPE_CHECKING, Any

if TYPE_CHECKING:
    from axion.client import Axion


class MemoryResource:
    def __init__(self, client: "Axion") -> None:
        self._c = client

    def ingest(
        self,
        content: str,
        *,
        kind: str = "semantic",
        importance: float = 0.5,
        source: str | None = None,
        metadata: dict[str, Any] | None = None,
        agent_id: str | None = None,
    ) -> dict:
        return self._c._request(
            "POST",
            "/memory/ingest",
            json={
                "content": content,
                "kind": kind,
                "importance": importance,
                "source": source,
                "metadata": metadata or {},
                "agent_id": agent_id,
            },
        )

    def recall(
        self,
        query: str,
        *,
        k: int = 8,
        kind: str | None = None,
        min_importance: float = 0.0,
    ) -> list[dict]:
        return self._c._request(
            "POST",
            "/memory/recall",
            json={
                "query": query,
                "k": k,
                "kind": kind,
                "min_importance": min_importance,
            },
        )

    def export(self, *, kind: str | None = None, limit: int = 10000) -> list[dict]:
        params: dict[str, Any] = {"limit": limit}
        if kind:
            params["kind"] = kind
        return self._c._request("GET", "/memory/export", params=params)

File: legacy/sdk/typescript/src/resources/memory.ts
/** Memory resource. */
import type { Axion } from "../client.js";
import type { MemoryEntry } from "../types.js";

export class MemoryResource {
  constructor(private readonly _c: Axion) {}

  ingest(opts: {
    content: string;
    kind?: string;
    importance?: number;
    source?: string;
    metadata?: Record<string, unknown>;
    agentId?: string;
  }): Promise<{ id: string; kind: string }> {
    return this._c._request("POST", "/memory/ingest", {
      body: {
        content: opts.content,
        kind: opts.kind ?? "semantic",
        importance: opts.importance ?? 0.5,
        source: opts.source,
        metadata: opts.metadata ?? {},
        agent_id: opts.agentId,
      },
    });
  }

  recall(opts: {
    query: string;
    k?: number;
    kind?: string;
    minImportance?: number;
  }): Promise<MemoryEntry[]> {
    return this._c._request<MemoryEntry[]>("POST", "/memory/recall", {
      body: {
        query: opts.query,
        k: opts.k ?? 8,
        kind: opts.kind,
        min_importance: opts.minImportance ?? 0,
      },
    });
  }

  export(opts: { kind?: string; limit?: number } = {}): Promise<MemoryEntry[]> {
    return this._c._request<MemoryEntry[]>("GET", "/memory/export", {
      query: { kind: opts.kind, limit: opts.limit ?? 10000 },
    });
  }
}
